File: tslang/test/hashmap/left-join/left-join.test.ts
import { describe, expect, test } from "bun:test";
import { leftJoin } from "../../../src/hashmap/left-join/left-join";

describe("test left join function", () => {
	test("test with map", () => {
		const hashMapOne = new Map([
			["diligent", "employed"],
			["font", "enamored"],
			["guide", "usher"],
			["outfit", "garb"],
			["wrath", "anger"],
		]);

		const hashMapTwo = new Map([
			["diligent", "idle"],
			["font", "averse"],
			["guide", "follow"],
			["flow", "jam"],
			["wrath", "delight"],
		]);

		const hashMapResult = [
			["diligent", "employed", "idle"],
			["font", "enamored", "averse"],
			["guide", "usher", "follow"],
			["outfit", "garb", null],
			["wrath", "anger", "delight"],
		];

		expect(leftJoin(hashMapOne, hashMapTwo)).toEqual(hashMapResult);
	});
});

File: golanglang/test/hashmap/left-join/left-join_test.go
package leftjoin_test

import (
	"reflect"
	"testing"

	leftjoin "github.com/1ChristianAlex/code-studies/internal/hashmap/left-join"
)

func TestDoLeftJoin(t *testing.T) {
	hashMapOne := map[string]string{
		"diligent": "employed",
		"font":     "enamored",
		"guide":    "usher",
		"outfit":   "garb",
		"wrath":    "anger",
	}

	hashMapTwo := map[string]string{
		"diligent": "idle",
		"font":     "averse",
		"guide":    "follow",
		"flow":     "jam",
		"wrath":    "delight",
	}

	hashMapResult := [][]string{
		{"diligent", "employed", "idle"},
		{"font", "enamored", "averse"},
		{"guide", "usher", "follow"},
		{"outfit", "garb", ""},
		{"wrath", "anger", "delight"},
	}

	joinResult := leftjoin.DoLeftJoin(hashMapOne, hashMapTwo)

	if !reflect.DeepEqual(joinResult, hashMapResult) {
		t.Fatalf("Test fail")
	}
}
